design/gateway-apis-implementation.md

```diff
@@ -62,69 +61,77 @@ Project Contour (the organisation) may investigate a Gateway API based layer 4 s
 
 ### Contour Gateway model
 
-Contour considers a Gateway to describe a single Envoy deployment, deployed and managed by something outside of itself.
-Contour expects to be supplied with the full name (name and namespace) of a single Gateway, which it will watch and update the status of.
-This will be a configuration file parameter.
-When a Gateway is not supplied, Contour will not do Gateway API processing, or in other words, the existing `--experimental-service-apis` flag will be obsoleted and removed).
-Inside Contour will merge Listeners within its Gateway. (See the detailed design for the exact rules Contour will use for this.)
+Contour considers a Gateway to describe a single Envoy deployment.
+Contour expects to be supplied with the full name and namespace of a single Gateway as well as the GatewayClass controller name.
```

Comment:
I forget the conversation on this, are we still going to require the gateway name/namespace or are we phasing that out for just the controller name? I vaguely remember something about how we choose which Gateway to admit following the conflict resolution guidelines in the spec etc. but I don't remember where that got written down

Reply:
> I forget the conversation on this, are we still going to require the gateway name/namespace or are we phasing that out for just the controller name?

It was my understanding that the gateway ns/name config will be removed and Contour will reconcile all gateway's that reference a gatewayclass matching its `controllerName`.

Reply:
But there's no way to specify a single controller in the gatewayclass. What happens when multiple gateways match the same controller? 

```
$ kubectl get gatewayclass,gateway -A                                                                                 
NAME                                                  CONTROLLER                             AGE
gatewayclass.networking.x-k8s.io/contour-class        projectcontour.io/ingress-controller   2m47s
gatewayclass.networking.x-k8s.io/contour-class-dupe   projectcontour.io/ingress-controller   2m47s

NAMESPACE        NAME                                       CLASS                AGE
projectcontour   gateway.networking.x-k8s.io/contour        contour-class        2m24s
projectcontour   gateway.networking.x-k8s.io/contour-dupe   contour-class-dupe   20s
```

Reply:
I guess the suggestion is that the controllername is a convention matching the contour namespace/name. We can do that, just don't think folks are going to get this right.